### ISSUE TITLE
os download: should not require '.prod' suffix for ESR versions

### ISSUE THREAD

**Issue by pdcastro** [balena-io/balena-cli#2387]
In the case of non-ESR images, `os download` defaults to production images when the `'.prod'` suffix is not added to the version string:

```
$ balena os download intel-nuc -o os.img --version 2.83.18+rev1
Getting device operating system for intel-nuc
Downloading balenaOS version 2.83.18+rev1.prod [                        ] 1% eta 2m28s

$ balena os download intel-nuc -o os.img --version 2.83.18+rev1.prod
Getting device operating system for intel-nuc
Downloading balenaOS version 2.83.18+rev1.prod [                        ] 1% eta 3m13s
```

But in the case of ESR images, it fails unless `'.prod'` is added:

```sh
$ balena os download intel-nuc -o os.img --version 2021.10.0
Getting device operating system for intel-nuc
BalenaRequestError: Request error: "Server error"

$ balena os download intel-nuc -o os.img --version 2021.10.0.prod
Getting device operating system for intel-nuc
Downloading balenaOS version 2021.10.0.prod [=                       ] 3% eta 1m19s
```

Affects CLI v12.52.0, which added support for ESR versions.

While at it: `balena os download` should also add ESR usage examples to the `balena help os download` output.

**Comment by floion** [comment]
This also happened on rpi4 with OS version `2.88.4`. Needed to append `.prod` or `.dev` to get it download the OS image successfully.

**Comment by pdcastro** [comment]
Resolved in CLI v12.55.3 or later.
